menpofit/io.py

```diff
@@ -150,9 +232,25 @@ def fit_from_bb(self, image, bounding_box, **kwargs):
         final_kwargs = self._fit_from_bb_kwargs.copy()
         # If the user provided kwargs at runtime, they take precedence.
         final_kwargs.update(kwargs)
-        # call the wrapped fitter with the updated kwargs
-        return self.wrapped_fitter.fit_from_bb(image, bounding_box,
-                                               **final_kwargs)
+        # check if pre-processing function exists
+        if self._image_preprocess is None:
+            # call the wrapped fitter with the updated kwargs
+            result = self.wrapped_fitter.fit_from_bb(image, bounding_box,
+                                                     **final_kwargs)
+        else:
+            # pre-process the image
+            proc_image, trans = self._image_preprocess(image, bounding_box)
+            # call the wrapped fitter with the updated kwargs
+            result = self.wrapped_fitter.fit_from_bb(
+                proc_image, proc_image.landmarks['__pointcloud'].lms,
```

Comment:
I think we should avoid attaching this `__pointcloud` magic key - we don't need it. Instead, 
```py
proc_image, trans = self._image_preprocess(image, bounding_box)
self.wrapped_fitter.fit_from_bb(
    proc_image, trans.apply(bounding_box),  **final_kwargs)